### PR TITLE
add sagemaker-studio-image-build

### PR DESCRIPTION
**Pull request by raybellwaves**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by raybellwaves** [comment]
`conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform linux-64: {MatchSpec("sagemaker[version='<3.0']")}
`

`- nothing provides requested sagemaker <3.0`

**Comment by raybellwaves** [comment]
`conda_build.exceptions.DependencyNeedsBuildingError: Unsatisfiable dependencies for platform win-64: {MatchSpec("sagemaker-python-sdk[version='<3.0']")}

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=438474&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=06d62fd2-ab49-5add-9ee5-83b6882d7d2b&l=839

Not sure If I can do more here

**Comment by raybellwaves** [comment]
don't think there a way to skip window in CI here but recipe has specs for not win

https://github.com/raybellwaves/staged-recipes/blob/add-sagemaker-studio-image-build/recipes/sagemaker-studio-image-build/meta.yaml#L14-L23

File: recipes/sagemaker-studio-image-build/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+source:
+  url: https://pypi.io/packages/source/{{ name[0] }}/{{ name }}/sagemaker_studio_image_build-{{ version }}.tar.gz
+  sha256: b7586597ca2395a63e463f8e3ab981ae987d5ea500bf86096af9a79dc41ec9b5
+
+build:
+  number: 0
+  entry_points:
+    - sm-docker=sagemaker_studio_image_build.cli:main
+  script: {{ PYTHON }} -m pip install . -vv
+  skip: true  # [not win]
```
Comment (dopplershift) [inline recipes/sagemaker-studio-image-build/meta.yaml:18]: Right now this says "Skip everything EXCEPT Windows", so I don't think that's what you want?

Comment (raybellwaves) [reply recipes/sagemaker-studio-image-build/meta.yaml:18]: whoops. What's the syntax for the opposite: skip windows ?
copying https://github.com/conda-forge/sagemaker-python-sdk-feedstock/blob/master/recipe/meta.yaml#L17
does the text after the # do anything? For exaple sagemaker-python-sdk has # [win]

Comment (dopplershift) [reply recipes/sagemaker-studio-image-build/meta.yaml:18]: Yes, after the `#` is the selector, that says what conditions `skip: true` applies to. So if you want to skip Windows, then it should be `skip: true  # [win]`

Comment (raybellwaves) [reply recipes/sagemaker-studio-image-build/meta.yaml]: Cheers. I naively assumed it was a comment

File: recipes/sagemaker-studio-image-build/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+  skip: true  # [win]
+
+requirements:
+  host:
+    - pip
+    - python
+  run:
+    - boto3 >=1.10.44,<2.0
+    - pandas
+    - pathos
```
Comment (dopplershift) [inline recipes/sagemaker-studio-image-build/meta.yaml:27]: I don't see either of these in any of the code for the tarball for this.

Comment (raybellwaves) [reply recipes/sagemaker-studio-image-build/meta.yaml:27]: thanks for digging in here.

See https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=439070&view=logs&j=6f142865-96c3-535c-b7ea-873d86b887bd&t=22b0682d-ab9e-55d7-9c79-49f3c3ba4823&l=1233
Looks as though the build fails without these. Any suggestions?

Comment (raybellwaves) [reply recipes/sagemaker-studio-image-build/meta.yaml:27]: Make a PR to sagemaker-python-sdk? I see pathos missing there in a recent PR by autotick bot https://github.com/conda-forge/sagemaker-python-sdk-feedstock/pull/127#issue-1088670416

**Comment by raybellwaves** [comment]
waiting on https://github.com/conda-forge/staged-recipes/pull/17488

**Review by dopplershift — APPROVED** [review]

**Pull request merged**